src/broker.js

```diff
@@ -0,0 +1,100 @@
+'use strict';
+
+var checkTypes = require('check-types'),
+	logger = require('./logger'),
+	traverson = require('traverson-promise'),
+	JsonHalAdapter = require('traverson-hal');
+
+// register the traverson-hal plug-in for media type 'application/hal+json'
+traverson.registerMediaType(JsonHalAdapter.mediaType, JsonHalAdapter);
+
+var pactURLPattern = "/pacts/provider/%s/latest",
+	pactURLPatternWithTag = "/pacts/provider/%s/latest/%s";
+
+// Constructor
+function Broker(provider, brokerUrl, tags, username, password) {
+	this._options = {};
+	this._options.brokerUrl = brokerUrl;
+	this._options.provider = provider;
+	this._options.tags = tags;
+	this._options.username = username;
+	this._options.password = password;
+};
+
+// Find Pacts returns the raw response from the HAL resource
+Broker.prototype.findPacts = function (tag) {
+	logger.debug("finding pacts: ", tag)
+	var linkName = (tag) ? 'pb:latest-provider-pacts-with-tag' : 'pb:latest-provider-pacts';
+	return traverson
+		.from(this._options.brokerUrl)
+		.withTemplateParameters({ provider: this._options.provider, tag: tag })
+		.withRequestOptions(this.getRequestOptions())
+		.jsonHal()
+		.follow(linkName)
+		.getResource()
+		.result
+};
+
+// Find all consumers collates all of the pacts for a given provider (with optional tags)
+// and removes duplicates (e.g. where multiple tags on the same pact)
+Broker.prototype.findConsumers = function () {
+	logger.debug("Finding consumers")
+
+	var promises = (this._options.tags.length > 0) ? this._options.tags.map(this.findPacts, this) : [this.findPacts()];
+
+	return Promise
+		.all(promises)
+		.then(function (values) {
+			var pactUrls = {};
+			values.forEach(function (response) {
+				response._links.pacts.forEach(function (pact) {
+					pactUrls[pact.title] = pact.href;
+				})
+			});
+			return Object.keys(pactUrls).reduce(function (pacts, key) {
+				pacts.push(pactUrls[key])
+				return pacts
+			}, [])
+		})
+};
+
+Broker.prototype.getRequestOptions = function () {
+	if (this._options.username && this._options.password) {
```

Comment:
does this object have to be 'built' at runtime?  Seems like it would be static and would never change.  I think it should be built within the constructor.